### ISSUE TITLE
[$500] Web - IOU - "Submit" button does not work correctly

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#33464]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.15-1
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
1. Go to staging.new.expensify.com
2. Login with Admin and Employee on two different browsers
3. From Employee's Workspace chat create Money request
4. Click on Submit button
5. On created IOU click Submit from Admin account

## Expected Result:
When Employee click Submit there should be message that IOU is summited and Submit button should be disabled

## Actual Result:
"Submit" button does not work correctly.
When employee click on it the button is still present and active
When admin click on it - in flash button turn to pay elsewhere and than again turn to Submit

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] Windows: Chrome 
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/7b454391-d14b-4728-833a-6190a30bd16b

 
[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01e924eea1632e9267</li>
        <li>Upwork Job ID: 1737947576594128896</li>
        <li>Last Price Increase: 2024-04-23</li>
    </ul>
</details>

**Comment by ikevin127** [comment]
**Note**: Feel free to use the information below to come up with a proposal.

### Debugging observations:
- the Submit buttons appear only for expense reports

- the issue happens because the `SubmitReport` call fails on both admin / employee sides with the following response:
```json
{
    "code": 666,
    "jsonCode": 666,
    "type": "Expensify\\Error\\Policy\\PolicyHasOwedAmount",
    "UUID": "E7E15357-F73C-42A0-B9B7-2A891A481044",
    "message": "Please have the policy's billing owner (kevin.bader96+1@gmail.com) visit the settings page and update their billing information.",
    "title": "Policy billing is out of date",
    "data": {
        "reportID": 1586586733220844,
        "isOwner": false,
        "ownerEmail": "kevin.bader96+1@gmail.com"
    },
    "htmlMessage": "Please have the policy's billing owner (kevin.bader96+1@gmail.com) visit the settings page and update their billing information.",
    "onyxData": [],
    "requestID": "839442b50874b9cb-OTP"
}
```
- the Submit button on the admin's side is grayed out because `props.chatReport.isOwnPolicyExpenseChat` is `false`

- the Submit button on the employee's side disappears and re-appears because while `SubmitReport` call is happening ReportPreview.js's `shouldShowSubmitButton` becomes `false` briefly until the call is finalized

- the Submit button on the admin's side becomes the green Pay with Expensigy button because while `SubmitReport` call is happening ReportPreview.js's `shouldShowSubmitButton` becomes `false` briefly while `shouldShowSettlementButton` becomes `true` briefly until the call is finalized

#### I don't want to invest time working on this issue but if I were to do it I would attempt fixing the issue by:
- displaying the `SubmitReport` call fail response error in the UI
- adding the `isDisabled` prop to the button in case there is a back-end error like the one mentioned above
- allow the user to clear the error which in turn would enable the Submit button, as the user was told why the action failed
- working on the Submit / Pay with Expensify buttons logic that trigger the brief hidding / switch while the `SubmitReport` call is happening, in order to fix the bad UX

**Comment by cubuspl42** [comment]
We're waiting on proposals

**Comment by shubham1206agra** [comment]
@cubuspl42 This may be fixed with https://github.com/Expensify/App/pull/33811

**Comment by sakluger** [comment]
@cubuspl42 do you think that PR will fix this issue? If so, I can put this one on hold.

**Comment by sakluger** [comment]
I'll go ahead and put it on hold for now. We can always take the hold off if it's not connected.

**Comment by sakluger** [comment]
I just looked at the bug report again and I think the real issue is that the admin saw a submit button at all to begin with. After testing again, I'm not able to reproduce - the money request immediately says has a "Pay with Expensify" button for the admin.

I'm going to close this out as non-reproduceable, feel free to bring back up for discussion in Slack if it's still an issue.

**Comment by lanitochka17** [comment]
Issue is still reproducible on the latest build 1.4.63-0


https://github.com/Expensify/App/assets/78819774/4f5571a0-c1d6-454d-a669-8002ebb179eb

**Comment by mallenexpensify** [comment]
@DylanDylann reassigning, please take over as C+. If you don't have bandwidth, unassign yourself. Thanks

**Comment by sakluger** [comment]
Waiting for proposals.

**Comment by sakluger** [comment]
Whoops, forgot to remove the hold in the issue title.

Hopefully now that it doesn't say [HOLD], we'll get some proposals.

**Comment by nkdengineer** [comment]
When admin click submit button, the BE return this error
 ```
jsonCode: 666
message: "This report must be submitted/forwarded/approved to the correct person according to the policy settings."
title: "Workflow Violation"
type: "Expensify\\Libs\\Error\\ExpError"
```

We need to ask an internal engineer to verify the expected and the current logic from BE side

**Comment by DylanDylann** [comment]
Agreed. 

🎀 👀 🎀 C+ reviewed

**Comment by DylanDylann** [comment]
@rlinoz Could you help to check [this comment](https://github.com/Expensify/App/issues/33464#issuecomment-2084762822)?

**Comment by kmbcook** [comment]
This problem is happening to me. I can't get around it. I created a collect workspace in old dot. I was on a free trial, so I could do development. Then the free trial ended. Now I have deleted all my old collect workspaces and old dot says that I have an outstanding balance. I have never entered any billing info. How can I owe Expensify any money? I just created a new workspace in new dot, but it still has the same problem. I can't submit any expenses. When I look at the server response from trying to submit it says that my billing info needs to be updated. Perhaps this information can help to solve this issue. Also, I need to get over this hump to be able to do development. Can anyone help me with this problem?

**Comment by sakluger** [comment]
@kmbcook your outstanding balance issue sounds unrelated to this issue - it sounds like a standard subscription payment issue. Could you please report this issue to Concierge, either by chatting with Concierge in NewDot or by emailing concierge@expensify.com? Our customer support team should be able to help.

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by sakluger** [comment]
If the issue is not reproduceable, then we should close the issue. Please reopen (or tag me in Slack if you need me to reopen it).
